app/controllers/submissions_controller.rb

```diff
@@ -20,22 +20,26 @@ def create
       return
     end
 
+    submission = Submission.new(
+      user_id: current_user.id,
+      problem_id: problem.id,
+      language: submission_lang
+    )
+
     ActiveRecord::Base.transaction do
       file = DbFile.create(
         name: uploaded_file.original_filename,
         content_type: 'text/plain',
-        contents: uploaded_file.read,
-      )
-
-      submission = Submission.create(
-        user_id: current_user.id,
-        problem_id: problem.id,
-        db_file_id: file.id,
-        language: submission_lang
+        contents: uploaded_file.read
       )
 
-      render status: :created, json: submission
+      submission.db_file_id = file.id
+      submission.save!
     end
+
+    RunSubmissionsJob.perform_later(submission.id)
+
+    render status: :created, json: submission
```

Comment:
Yeah, right now the transaction re-throws exceptions. `.save!` will throw an exceptions if it's not successful.